src/AstSemanticChecker.cpp

```diff
@@ -433,6 +433,18 @@ void AstSemanticChecker::checkLiteral(
 
 void AstSemanticChecker::checkAggregator(
         ErrorReport& report, const AstProgram& program, const AstAggregator& aggregator) {
+    const AstAggregator* inner = nullptr;
+
+    visitDepthFirst(aggregator, [&](const AstAggregator& innerAgg) {
+        if (aggregator != innerAgg) {
+            inner = &innerAgg;
+        }
+    });
+
+    if (inner) {
```

Comment:
Should change to:
```suggestion
    if (inner != nullptr) {
```
to be more consistent with the rest of the codebase.